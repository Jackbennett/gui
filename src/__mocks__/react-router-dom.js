import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
