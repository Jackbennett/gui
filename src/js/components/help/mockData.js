export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
