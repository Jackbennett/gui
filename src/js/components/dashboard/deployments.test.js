import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import Deployments from './deployments';
import { undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('Deployments Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      deployments: {
        byId: {},
        byStatus: {
          finished: { total: 0 },
          inprogress: { total: 0 },
          pending: { total: 0 }
        }
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Deployments getDeployments={() => Promise.resolve()} setSnackbar={jest.fn()} />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
