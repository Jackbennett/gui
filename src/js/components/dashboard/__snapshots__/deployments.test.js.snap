// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deployments Component renders correctly 1`] = `
<div>
  <h4
    className="dashboard-header"
  >
    <span>
      Deployments
    </span>
  </h4>
  <div
    className="deployments"
    style={
      Object {
        "marginBottom": 50,
      }
    }
  >
    <div
      className="loaderContainer"
    >
      <div
        className="  loader"
      >
        <span
          className="dot dot_1"
        />
        <span
          className="dot dot_2"
        />
        <span
          className="dot dot_3"
        />
        <span
          className="dot dot_4"
        />
      </div>
    </div>
  </div>
</div>
`;
